tuf/repository_lib.py

```diff
@@ -1485,12 +1485,23 @@ def generate_root_metadata(version, expiration_date, consistent_snapshot,
     roledict[rolename] = role_metadata
 
   # Generate the root metadata object.
-  root_metadata = tuf.formats.RootFile.make_metadata(version, expiration_date,
-                                                     keydict, roledict,
-                                                     consistent_snapshot,
-                                                     compression_algorithms)
-
-  return root_metadata 
+  # Use generalized build_dict_conforming_to_schema func to produce a dict that
+  # contains all the appropriate information for this type of metadata,
+  # checking that the result conforms to the appropriate schema.
+  # TODO: Later, probably after the rewrite for TUF Issue #660, generalize
+  #       further, upward, by replacing generate_targets_metadata,
+  #       generate_root_metadata, etc. with one function that generates
+  #       metadata, possibly rolling that upwards into the calling function.
+  #       There are very few things that really need to be done differently.
+  return tuf.formats.build_dict_conforming_to_schema(
+      tuf.formats.ROOT_SCHEMA,
+      _type='Root',   # TODO: Does this have to be capitalized? -.-
```

Comment:
Just a thought, would it make sense to make `_type` optional  in `build_dict_conforming_to_schema` and use the one defined in the schema object, if it isn't passed?

Reply:
On casing: the casing is at odds with the same content in theupdateframework/tuf.  The modern reference implementation mandates lower-cased _type.  I'm grumbling about that.

On _type being optional: I thought about something like that, but I'd like `build_dict_conforming_to_schema` to remain general enough to be used for every schema construction, and code handling `_type` specially feels like cheating.  If the long list of function call arguments becomes an issue, then it might be wise to do this.

Reply:
Re casing: I did have the feeling that your `# TODO`-question was rhetorical. :)

Re _type: You're right, I didn't think of other schemas. Though it might be okay to restrict this (or an additional wrapper) function to schemas that have a `_type` field (i.e. `{ROOT, TARGETS, SNAPSHOT, TIMESTAMP}_SCHEMA`). What do you think?

Note that the function as it currently is is already restricted to a subset of schemas, i.e. `Object`-type schemas...

```python
>>> from securesystemslib.formats import HEX_SCHEMA
>>> 
>>> HEX_SCHEMA.matches("beef")
True
>>>
>>> build_dict_conforming_to_schema(HEX_SCHEMA, "beef")
# [...]
TypeError: build_dict_conforming_to_schema() takes 1 positional argument but 2 were given
>>>
>>> build_dict_conforming_to_schema(HEX_SCHEMA, dead="beef")
# [...]
securesystemslib.exceptions.FormatError: {'dead': 'beef'} did not match 'pattern /[a-fA-F0-9]+$/'
```

Obviously not a blocker ... especially since the PR is already merged :) ... just some thoughts...